plugin/evm/vm.go

```diff
@@ -279,6 +280,12 @@ func (vm *VM) Initialize(
 		g.Config.FeeConfig = params.DefaultFeeConfig
 	}
 
+	if len(upgradeBytes) != 0 {
+		if err := vm.handleUpgradeBytes(upgradeBytes); err != nil {
```

Comment:
We should persist the parsed upgrade for each timestamp and error if it changes. This alone will save us MUCH pain.